libraries/AP_Proximity/AP_Proximity_AirSimSITL.cpp

```diff
@@ -34,15 +34,17 @@ void AP_Proximity_AirSimSITL::update(void)
 
     set_status(AP_Proximity::Status::Good);
 
-    memset(_distance_valid, 0, sizeof(_distance_valid));
+    for(uint8_t i=0; i < PROXIMITY_NUM_SECTORS; i++) {
+        boundary.mark_distance_valid(false, i);
+    }
 
     for (uint16_t i=0; i<points.length; i++) {
         Vector3f &point = points.data[i];
         if (point.is_zero()) {
             continue;
         }
         const float angle_deg = wrap_360(degrees(atan2f(point.y, point.x)));
-        const uint8_t sector = convert_angle_to_sector(angle_deg);
+        const uint8_t sector = boundary.convert_angle_to_sector(angle_deg);
```

Comment:
Boundary_3D::convert_angle_to_sector() should probably accept a pitch angle as well.  the sector returned should either be a new enum defined within Boundary_3D or a new structure or class (also defined within Boundary_3D) made up of a "stack" and "sector".

At the moment I think we are populating the boundary ignoring the pitch angle but I'm more and more convinced that that will need to change.